File: src/main/java/com/ptah/entity/userprofiling/ProjectNomination.java
package com.ptah.entity.userprofiling;

import com.ptah.common.BaseEntity;
import com.ptah.entity.project.Project;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class ProjectNomination extends BaseEntity {
    @ManyToOne
    private User user;
    @ManyToOne
    private Project project;
    @Enumerated(EnumType.STRING)
    private ProjectRole projectRole;
}

File: src/main/kotlin/com/ptah/entity/userprofiling/ProjectNomination.kt
package com.ptah.entity.userprofiling

import com.ptah.common.BaseEntity
import com.ptah.entity.project.Project
import jakarta.persistence.*
import lombok.*

@Entity
@Getter
@Setter
@Builder
class ProjectNomination : BaseEntity() {
    @ManyToOne
    private val user: User? = null

    @ManyToOne
    private val project: Project? = null

    @Enumerated(EnumType.STRING)
    private val projectRole: ProjectRole? = null
}
